src/map/skill.cpp

```diff
@@ -18243,6 +18243,12 @@ struct s_skill_condition skill_get_requirement(struct map_session_data* sd, uint
 				if (req.itemid[i] == ITEMID_MAGIC_GEAR_FUEL && sd->special_state.no_mado_fuel)
 					req.itemid[i] = req.amount[i] = 0;
 			}
+
+			// Process level_dependent requirement
+			if (level_dependent && skill_lv <= MAX_SKILL_ITEM_REQUIRE) {
+				req.itemid[0] = skill->require.itemid[skill_lv - 1];
+				req.amount[0] = skill->require.amount[skill_lv - 1];
+			}
```

Comment:
I agree with the fix, but there's just one issue with it's location. GitHub won't let me suggest the location of the fix, but it needs to be moved up because where you have it now, you're overwriting a few checks that remove item requirements through skill or item bonuses.

Suggested fix:
![image](https://user-images.githubusercontent.com/3521753/162455830-68d68798-d2ab-4d29-80ae-efb01f33823d.png)

Reply:
okay, thanks for you review.
I will push a new commit to adjust it after few mins.